### PR TITLE
Added support for pre and post commands

### PR DESCRIPTION
**Pull request by ghost**
Two new fields were added to the config. Precmds runs before cmds and
links. Postcmds runs after cmds and links. Cmds is no longer necessary
but was left in for compatibility.

### CODE REVIEW

**Comment by ghost** [comment]
This is useful for a variety of programs.

As an example one might use xmonad. First one must install xmonad, then link ones configuration, then after it has been installed one must ```xmonad --recompile``` to use the config.

File: task.go [hunk 1]
```diff
@@ -28,12 +28,14 @@ import (
...
 type task struct {
-	Deps    []string
-	Links   [][]string
-	Cmds    [][]string
-	Envs    [][]string
-	Accepts [][]string
-	Rejects [][]string
+	Deps     []string
+	Links    [][]string
+    Precmds  [][]string
```
Comment (FooSoft) [inline task.go]: Please use tabs, not spaces for indentation, rename `Precmds` to `CmdsPre` and `Postcmds` to `CmdsPost` to conform with naming convention.

File: task.go [hunk 2]
```diff
@@ -61,6 +63,14 @@ func (t *task) process(conf *config) error {
 		}
 	}
 
+    if conf.flags&flagNoCmds == 0 {
```
Comment (FooSoft) [inline task.go]: Combine the `if` check with the one directly below it, no need to perform the same test twice.

Comment (FooSoft) [reply task.go]: Also again, tabs not spaces

**Comment by FooSoft** [comment]
Please fix the issues I pointed out, retab the source file. It would be nice if you also updated the documentation to reflect these new features (mention them in the paragraph below):

> In addition to creating links, Homemaker is capable of executing commands on a per-task basis. Commands should be defined in an array called cmds, split into an item per each command line argument. All of the commands are executed with dest as the working directory (as mentioned previously, this defaults to your home directory). If any command returns a nonzero exit code, Homemaker will display an error message and prompt the user to determine if it should abort, retry, or cancel.

File: task.go (hunk 2 above)
Comment (FooSoft) [reply task.go]: You should just be able to run `gofmt` on the source file.

**Comment by ghost** [comment]
I fixed the issues you mentioned. I also edited the docs, althought I feel I might have changed the style of the entry. If you feel that is the case I will try to rewrite them in a style more consistent with the rest of the documentation.

**Comment by FooSoft** [comment]
It was close enough, so I merged it in and added some small tweaks. Thanks for the code.

**Pull request merged**